core/market/decentralized/src/db/dao/demand.rs

```diff
@@ -43,19 +44,29 @@ impl<'c> DemandDao<'c> {
         .await
     }
 
-    pub async fn get_demands_before(
+    pub async fn get_demands(
         &self,
-        insertion_ts: NaiveDateTime,
+        node_id: Option<NodeId>,
+        insertion_ts: Option<NaiveDateTime>,
         validation_ts: NaiveDateTime,
     ) -> DbResult<Vec<Demand>> {
         readonly_transaction(self.pool, move |conn| {
-            Ok(dsl::market_demand
-                // we querying less then here and less equal in Offers
-                // not to duplicate pair subscribed at the very same moment
-                .filter(dsl::insertion_ts.lt(insertion_ts))
+            let mut query = dsl::market_demand
                 .filter(dsl::expiration_ts.ge(validation_ts))
                 .order_by(dsl::creation_ts.asc())
-                .load::<Demand>(conn)?)
+                .into_boxed();
+
+            if let Some(ts) = insertion_ts {
+                // we querying less then here and less equal in Offers
+                // not to duplicate pair subscribed at the very same moment
```

Comment:
Outdated log??
`Offers`

Reply:
Altough I just moved this line :) it still looks relevant:
https://github.com/golemfactory/yagna/blob/3f8f7f9d7472e2bbe44076636eab58a2c1889958/core/market/decentralized/src/db/dao/offer.rs#L88-L90

Reply:
Ok, but there's `Offer` and we are dealing with `Demands`